systems.json gets automaticly downloaded

Yvonne-Aizawa opened the pull request:
In world explorer systems.json gets automatically downloaded when it does not exist yet. 

this fixes #15

Yvonne-Aizawa committed:
Contract info tab added

thats it really. It works the same way the ship info box works.

Yvonne-Aizawa committed:
ran cargo fmt

Yvonne-Aizawa committed:
explore the universe (#1)

added a universe explorer tab. with it you can look at systems and waypoints. you can set to only see waypoints that have your ships.

Yvonne-Aizawa committed:
refuel refreshes ship but still need to re select ship in list

Yvonne-Aizawa committed:
dark and light mode

Yvonne-Aizawa committed:
hopefully fixed merge conflict errors

Yvonne-Aizawa committed:
Working again

Yvonne-Aizawa committed:
Merge branch 'develop' into master

Yvonne-Aizawa committed:
now it should work :)

i suck at merge requests

Yvonne-Aizawa committed:
ship_info.rs had duplicate code

Yvonne-Aizawa committed:
Merge branch 'Allstreamer:develop' into master

Yvonne-Aizawa committed:
updated spacedust to 1.0.6

added email option in auth tab
backend now takes an email string to register

ship info. seems that ship faction symbol is a string now. (was wrapped in option before)

Yvonne-Aizawa committed:
ran cargo fmt

Yvonne-Aizawa committed:
check if mail field is filled

Yvonne-Aizawa committed:
Merge branch 'Allstreamer:develop' into master

Yvonne-Aizawa committed:
resovled issue #15

the systems.json file gets automatically downloaded when it does not exist yet

Yvonne-Aizawa committed:
Merge branch 'master' of https://github.com/Yvonne-Aizawa/alltrader

Allstreamer commented on trader/src/parse_system.rs:
```diff
@@ -37,16 +39,49 @@ pub fn parse_json() -> Result<Vec<System>, std::io::Error> {
+                    println!("File downloaded successfully");
+                    let mut file = File::open(file_path).unwrap();
+                    let mut contents = String::new();
+                    file.read_to_string(&mut contents)
+                        .expect("Failed to read file");
+                    let systems: Vec<System> =
+                        serde_json::from_str(&contents).expect("Failed to parse JSON");
```
These two expects should probably be replaced with a `.wrap_err("")?` operator

change the type the function is returning to `color_eyre::Result<Vec<System>>` and it should allow you to return any error types (color_eyre is like anyhow if you know that crate)

Yvonne-Aizawa committed:
proper error handeling with color_eyre

there is a clippy warning however in backend.rs on line 133 that i do not know how to fix yet

Yvonne-Aizawa committed:
removed clippy warning

and ran cargo fmt this time

Yvonne-Aizawa reviewed (commented):
seems correct to me (since i wrote it myself)

Allstreamer commented:
One sec, gonna make some minor touch ups myself.

Allstreamer committed:
Overhaul Error handling

Allstreamer reviewed (approved):
Now errors don't panic the program but still log!
